qiskit_nature/algorithms/ground_state_solvers/oovqe_algorithm.py

```diff
@@ -0,0 +1,631 @@
+            name = problem.main_property_name
+            main_second_q_op = second_q_ops.pop(name, None)
+            if main_second_q_op is None:
+                raise ValueError(
+                    f"The main `SecondQuantizedOp` associated with the {name} property cannot be "
+                    "`None`."
+                )
+            aux_second_q_ops = second_q_ops
+
+        main_operator = self._qubit_converter.convert(
+            main_second_q_op,
+            num_particles=problem.num_particles,
+            sector_locator=problem.symmetry_sector_locator,
+        )
+        aux_ops = self._qubit_converter.convert_match(aux_second_q_ops)
+
+        if aux_operators is not None:
+            wrapped_aux_operators: ListOrDict[Union[SecondQuantizedOp, PauliSumOp]] = ListOrDict(
+                aux_operators
+            )
+            for name, aux_op in iter(wrapped_aux_operators):
+                if isinstance(aux_op, SecondQuantizedOp):
+                    converted_aux_op = self._qubit_converter.convert_match(aux_op, True)
+                else:
+                    converted_aux_op = aux_op
+                if isinstance(aux_ops, list):
+                    aux_ops.append(converted_aux_op)
+                elif isinstance(aux_ops, dict):
+                    if name in aux_ops.keys():
+                        raise QiskitNatureError(
+                            f"The key '{name}' is already taken by an internally constructed "
+                            "auxliliary operator! Please use a different name for your custom "
+                            "operator."
+                        )
+                    aux_ops[name] = converted_aux_op
+
+        # if the eigensolver does not support auxiliary operators, reset them
+        if not self._solver.supports_aux_operators():
+            aux_ops = None
+
+        return main_operator, aux_ops
+
+    def rotate_orbitals(self, matrix_a, matrix_b):
+
+        problem = copy.copy(self.problem)
+        grouped_property_transformed = problem.grouped_property_transformed
+
+        # use ElectronicBasisTransform
+        transform = ElectronicBasisTransform(ElectronicBasis.MO, ElectronicBasis.MO, matrix_a, matrix_b)
+
+        # only 1 & 2 body integrals have the "transform_basis" method,
+        # so I access them through the electronic energy
+        ee = grouped_property_transformed.get_property('ElectronicEnergy')
+        one_body_integrals = ee.get_electronic_integral(ElectronicBasis.MO, 1)
+        two_body_integrals = ee.get_electronic_integral(ElectronicBasis.MO, 2)
+
+        # the basis transform should be applied in place, but it's not???
+        # unless I manually add the integrals, the result of second_q_ops()
+        # doesn't change.
+        # I have to look further into this.
+        ee.add_electronic_integral(one_body_integrals.transform_basis(transform))
+        ee.add_electronic_integral(two_body_integrals.transform_basis(transform))
```

Comment:
@mrossinek I am a bit confused about this, but I had to explicitly add the electronic integrals to the electronic energy grouped property after applying transform basis, or else the output from` second_q_ops()` wouldn't change... but they are supposed to happen in place, right?

Reply:
Ok, they do not happen in place. Problem solved :)

Reply:
Ok, they do not happen in place. Problem solved :)

Reply:
Sorry for the lack of reply, I overlooked this mention notification! :see_no_evil: